spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/service/impl/DefaultTaskExecutionService.java

```diff
@@ -264,9 +264,18 @@ private TaskLauncher findTaskLauncher(String platformName) {
 					StreamSupport.stream(launcherRepository.findAll().spliterator(), false)
 							.map(Launcher::getName)
 							.collect(Collectors.toList());
-			throw new IllegalStateException(String.format("No Launcher found for the platform named '%s'.  " +
-							"Available platform names are %s",
-					platformName, launcherNames));
+			if (launcherNames.size() == 1) {
```

Comment:
I think we should fail fast when the platform name is explicitly given and it doesn't exist. Instead of always using the single platform even if the platform doesn't exist with the provided platform name, I think we need to use the single platform only if the platform name is not given explicitly. We probably need to fix our code [here](https://github.com/spring-cloud/spring-cloud-dataflow/blob/master/spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/service/impl/DefaultTaskExecutionService.java#L161)

Reply:
Whatever we end up deciding, let's please also make sure Skipper (for streams) follow the same pattern. We absolutely don't want to have a different platform-resolution behavior between streams vs. tasks.

Reply:
In a single platform with the name other than `default`, Skipper inserts a new deployer named `default` which means we would have the same platform named `default` and the actual name. We can either change this logic in Skipper or change the behavior here in this PR.

Reply:
I'd vote for making the change here in this PR for consistency. @markpollack: What are your thoughts?

Reply:
I don't remember why we picked that strategy in skipper, but consistency is a good argument.

Reply:
Was about to merge, but found a reason why we have the `default` platform being created.

There are some invocations to fetch the details about task executions use `default` as the platform name if none is provided. For instance, getLog() is one such case. I think we need to apply the same logic in those places as well.